server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4516,6 +4537,35 @@ public void onAfter() {
                     flushOrRollRunning.compareAndSet(true, false);
                 }
             }
+        } else if (shouldRefreshShard() && isRefreshRunning.compareAndSet(false, true)) {
+
+            if (shouldRefreshShard()) {
+                logger.info("submitting async Refresh request");
+                final AbstractRunnable _refresh = new AbstractRunnable() {
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.warn("forced refresh failed after number of uncommited translog files breached limit", e);
+                    }
+
+                    @Override
+                    protected void doRun() throws Exception {
+                        refresh("Too many uncommited Translog files");
+                    }
+
+                    @Override
+                    public boolean isForceExecution() {
+                        return true;
+                    }
+
+                    @Override
+                    public void onAfter() {
```

Comment:
Does this run always if the task execution starts?

Reply:
Yes, will always be called after the completion or rejection of task.